src/Dependencies/Collections/SegmentedDictionary`2.cs

```diff
@@ -1178,13 +1178,13 @@ public bool MoveNext()
                 return false;
             }
 
-            public KeyValuePair<TKey, TValue> Current => _current;
+            public readonly KeyValuePair<TKey, TValue> Current => _current;
 
-            public void Dispose()
+            public readonly void Dispose()
             {
             }
 
-            object? IEnumerator.Current
+            readonly object? IEnumerator.Current
```

Comment:
❔ What's the point of this, considering there is no way to directly invoke this member such that it will benefit from the `readonly` designation?

Reply:
marks intent that it should not mutate state.  can go along with anohter analyzer (or the compiler) that tells you if you violate this and actually mutate this guy (or cause it to be copied).

Reply:
I'm concerned this may add noise and/or confusion for people that don't understand how interface dispatch works today. In almost all cases where an explicitly interface implementation is called, the member is already operating on a copy that was created for that one call. Exceptions to this include:

1. The user is operating on a storage location which is statically-typed to the interface and not to the value type. In this case, the use of `readonly` isn't apparent to the caller at all.
2. The user has gone to [strange lengths](https://github.com/dotnet/roslyn/blob/main/src/Dependencies/Collections/Internal/IEnumerableCalls.cs) to ensure a constrained call is made without copying/boxing a value.

I see arguments for both keeping this behavior and skipping this behavior. Maybe discuss with the team before we merge?

Reply:
>  In almost all cases where an explicitly interface implementation is called, the member is already operating on a copy that was created for that one call. Exceptions to this include:

I actually disagree with this.  :) Passing a value type to a generic that is constrained on an interface is a normal pattern, and we expect to push it further in C#.  

The purpose of readonly here is important as a documentation/protection mechanism (similar to 'static' on a lambda).  The presence of it means no chance of mutation happening in the future (similar to how 'static' prevents capture in the future).  

I can bring to team to see.  But this seems like a pure benefit.  I'm not sure what could be bad/undesirable about marking these members this way :)